### PR TITLE
Bugfix/delete folders during propagation even when propagation has errors

### PR DESCRIPTION
**Pull request by mgallien**
do not skip folders to delete in case of errors with blacklisted files

clean up and log improvements

### CODE REVIEW

File: src/libsync/owncloudpropagator.h [hunk 1]
```diff
@@ -382,6 +382,8 @@ private slots:
 private:
 
     bool scheduleDelayedJobs();
+
+    SyncFileItem::Status _hasError = SyncFileItem::Status::NoStatus;
```
Comment (allexzander) [inline src/libsync/owncloudpropagator.h:386]: @mgallien I think the naming is misleading. I would expect it to be a bool member while it is an enum type. Could you name it something like `_errorStatus`?

File: src/libsync/owncloudpropagator.cpp [hunk 2]
```diff
@@ -1338,11 +1339,37 @@ void PropagateRootDirectory::slotSubJobsFinished(SyncFileItem::Status status)
...
+    }
+
     propagator()->scheduleNextJob();
 }
 
 void PropagateRootDirectory::slotDirDeletionJobsFinished(SyncFileItem::Status status)
 {
+    if (_hasError != SyncFileItem::NoStatus && status == SyncFileItem::Success) {
+        qCInfo(lcPropagator) << "PropagateRootDirectory::slotDirDeletionJobsFinished" << "reporting previous error" << _hasError;
+        status = _hasError;
```
Comment (allexzander) [inline src/libsync/owncloudpropagator.cpp:1370]: @mgallien What is the concept behind this? I am not sure I fully understand. Could you elaborate?

**Comment by allexzander** [comment]
@mgallien What exactly the issue is and how I can reproduce it?

File: src/libsync/owncloudpropagator.cpp (hunk 2 above)
Comment (mgallien) [reply src/libsync/owncloudpropagator.cpp:1370]: so if you look at the new automated test, you can see that we now check that deleted folders are deleted even if there are blacklisted errors during propagation
the idea is that we still want to do the folder deletions part of the propagation and if that part is succeeding, we still want to report a sync error at the end

**Review by allexzander — APPROVED** [review]

**Comment by allexzander** [comment]
@mgallien Please clean-up the history

**Pull request merged**